araproc/analysis/standard_reco.py

```diff
@@ -924,3 +926,364 @@ def get_avg_corr_snr(self, wave_packet, excluded_channels = []):
+                delay *= -1
+            
+            # Save the calculated arrival delay
+            delays[ch_ID] = delay
+
+        return delays
+
+    def get_csw(
+        self, wavepacket, is_software, is_calpulser, solution, polarization, reco_results,
+        excluded_channels, which_distance='distant'
+    ):
+        """
+        Build the Coherently Summed Waveform (CSW) for the given `useful_event`
+          by shifting each waveform accoriding to the time delay calculated 
+          after finding the max cross correlation between two channels 
+          in a window around the expected arrival delay. 
+
+        Parameters
+        ----------
+        data : AnalysisDataset
+        wavepacket : dict
+            AraProc wavepacket (keys include waveforms and the event number)
+        is_software : bool
+            Boolean for whether this event is a software trigger or not.
+        solution : int
+            0 for direct ray solution. 1 for reflected/refracted ray solution.
+        polarization : int
+            0 for VPol, 1 for HPol
+        reco_results : dict
+            Reco results already with the specific reconstruction reqeusted (so
+            the keys of this object include 'theta' and 'phi')
+        excluded_channels : set
+            Any channels to exclude that aren't already excluded by 
+            livetime configurations.
+        which_distance : str
+            The distance of the reconstruction for analysis, following the 
+            convention of other functions in this class. 
+
+        Returns
+        -------
+        csw : ROOT.TGraph
+            The coherently summed waveform. Only returned if `return_csw==True`
+        warning : int
+            If non-zero, contains information about something that went wrong
+              in the CSW calculation.
+        """
+
+        # Initialize warning object as 0
+        warning = 0
+
+        # Check that reco_results have been passed in properly
+        if 'theta' not in reco_results.keys():
+            raise ValueError(
+                "reco_result object passed into this function does not appear "
+                "to be for a specific reconstruction. Should have a 'theta' key "
+                "but only has the following:", reco_results.keys())
+
+        # Make the excluded_channels object a set to speed up computation
+        excluded_channels = set(excluded_channels)
+
+        # Determine which channels should be used in the csw
+        channels_to_csw = []
+        for ch_ID in wavepacket['waveforms'].keys(): 
+            if ch_ID not in excluded_channels:
+                channels_to_csw.append(ch_ID)
+
+        # Determine the "reference channel" to base the CSW around as the
+        #   channel with the maximum voltage
+        reference_ch = -123456
+        reference_ch_max_voltage = -1
+        for ch_ID in channels_to_csw:
+            this_max_voltage = np.max(wavepacket['waveforms'][ch_ID].GetY())
```

Comment:
also this one, wrap it in `np.asarray`. I'm too paranoid lol

Reply:
Yep, done!